File: Example/app/src/main/java/com/example/tengfei/example/MainActivity.java
package com.example.tengfei.example;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.example.tengfei.example.activity.ChatActivity;


/**
 * @author tengfei
 */
public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.tv01).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.tv01:
                skipActivity(ChatActivity.class);
                break;
            default:
                break;
        }
    }

    private <T> void skipActivity(Class<T> z) {
        Intent intent = new Intent(this, z);
        startActivity(intent);
    }
}

File: EasyIPC/app/src/main/java/com/example/easyipc/MainActivity.kt
package com.example.easyipc

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.example.ipclib.core.EasyBinderIPC

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //打开服务
        EasyBinderIPC.getInstance().open(this)


        //注册服务
        EasyBinderIPC.getInstance().register(ExampleSingle::class.java)

        ExampleSingle.getInstance().userInfo = UserInfo("tengfei","27")

    }

    fun skip(view: View) {

        val intent = Intent(this,SecondActivity::class.java)
        startActivity(intent)

    }
}